logstash-core/src/main/java/org/logstash/config/ir/compiler/EventCondition.java

```diff
@@ -363,8 +363,12 @@ private static EventCondition eq(final EventValueExpression first,
             final EventValueExpression second) {
             final FieldReference field1 = FieldReference.from(first.getFieldName());
             final FieldReference field2 = FieldReference.from(second.getFieldName());
-            return event -> event.getEvent().getUnconvertedField(field1)
-                .equals(event.getEvent().getUnconvertedField(field2));
+            return event -> {
+                Event java = event.getEvent();
+                return Objects.equals(
+                        java.getUnconvertedField(field1),
+                        java.getUnconvertedField(field2));
```

Comment:
Would a user expect that both fields being null would result in this returning true? 

(I think it can be argued both ways, but I'm curious on your thoughts - either way is better than an NPE, of course)

Reply:
My reasoning for the equality comparison being true when both fields are null is this:
- It matches the current behavior of the Ruby execution engine
- It matches both Ruby and Java semantics in which nil/null are equal to each other and there is a way to test if an individual field is nil if that is necessary.

I'd agree that there's a case to be made that configs should follow SQL-style semantics in which `NULL != NULL`, but I think that's probably a separate discussion since it could be a breaking change for some configs.

Reply:
Matching the semantics of the Ruby execution engine is the most important thing, so I'm +1 on `NULL == NULL`